[Bug]: randen_detect.cc compilation failure with Intel compilers

### Describe the issue

I decided to try compiling Tensorflow with the latest Intel compiler (/opt/intel/oneapi/compiler/2023.0.0). The compilation failed on abseil like so:

```
external/com_google_absl/absl/random/internal/randen_detect.cc:50:13: error: static declaration of '__cpuid' follows non-static declaration
static void __cpuid(int cpu_info[4], int info_type) {
            ^
external/com_google_absl/absl/random/internal/randen_detect.cc:50:13: note: '__cpuid' is a builtin
```

Apparently the Intel compilers define the __cpuid built-in similar to MSVC. (But the compiler itself is based on LLVM.)

### Steps to reproduce the problem

Compile abseil using any recent Intel compiler

### What version of Abseil are you using?

Just looking at the source code for randen-detect.cc this is present in the master branch.

### What operating system and version are you using?

Alma Linux 9.1

### What compiler and version are you using?

Intel compiler 2023.0.0 and 2022.2.1 both fail this way.

### What build system are you using?

N/A

### Additional context

Easiest fix is probably to detect the Intel compilers and do not define your own __cpuid() when using them.

The newer Intel compilers provide the preprocessor symbol __INTEL_LLVM_COMPILER ([see here for details](https://www.intel.com/content/www/us/en/developer/articles/technical/use-predefined-macros-for-specific-code-for-intel-dpcpp-compiler-intel-cpp-compiler-intel-cpp-compiler-classic.html)).

--- a/absl/random/internal/randen_detect.cc
+++ b/absl/random/internal/randen_detect.cc
@@ -45,6 +45,10 @@
 #if defined(ABSL_INTERNAL_USE_X86_CPUID)
 #if defined(_WIN32) || defined(_WIN64)
 #include <intrin.h>  // NOLINT(build/include_order)
+#elif ABSL_HAVE_BUILTIN(__cpuid)
+// MSVC-equivalent __cpuid intrinsic declaration for clang-like compilers
+// for non-Windows build environments.
+void __cpuid(int[4], int);
 #else
 // MSVC-equivalent __cpuid intrinsic function.
 static void __cpuid(int cpu_info[4], int info_type) {

